wsmaster/codenvy-hosted-api-license/src/main/java/com/codenvy/api/license/server/jpa/JpaCodenvyLicenseActionDao.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.api.license.server.jpa;
+
+import com.codenvy.api.license.model.Constants;
+import com.codenvy.api.license.server.dao.CodenvyLicenseActionDao;
+import com.codenvy.api.license.server.model.impl.CodenvyLicenseActionImpl;
+import com.google.inject.Singleton;
+import com.google.inject.persist.Transactional;
+
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.core.db.jpa.DuplicateKeyException;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * @author Anatolii Bazko
+ */
+@Singleton
+public class JpaCodenvyLicenseActionDao implements CodenvyLicenseActionDao {
+
+    @Inject
+    protected Provider<EntityManager> managerProvider;
+
+    @Override
+    public void store(CodenvyLicenseActionImpl codenvyLicenseAction) throws ServerException, ConflictException {
+        requireNonNull(codenvyLicenseAction, "Required non-null codenvy license action");
+
+        try {
+            doStore(codenvyLicenseAction);
+        } catch (DuplicateKeyException e) {
+            throw new ConflictException("Codenvy license action already exists");
+        } catch (RuntimeException e) {
+            throw new ServerException(e);
+        }
+    }
+
+    @Override
+    public void remove(Constants.License licenseType, Constants.Action actionType) throws ServerException {
+        doRemove(licenseType, actionType);
```

Comment:
Wrap `doRemove` call with `try catch` and rethrow `RuntimeException` as `ServerException` like defined by the interface.